201-front-door-create-caching/azuredeploy.json

```diff
@@ -32,12 +32,12 @@
         }
     },
     "variables": {
-        "frontdoorref": "[resourceId('Microsoft.Network/frontDoors', parameters('frontDoorName'))]",
-        "frontdoorLocation": "global"
+        "frontdoorLocation": "global",
+        "frontdoorSuffix": ".azurefd.net"
```

Comment:
this needs to be retrieved with the reference() function

Reply:
The resource cannot make a reference to itself while is being created.